source/braille.py

```diff
@@ -1398,18 +1406,23 @@ def getFocusContextRegions(obj, oldFocusRegions=None):
 			# No common regions were found.
 			commonRegionsEnd = 0
 			newAncestorsStart = 1
-		# Yield the common regions.
+		# Yield the common regions, thereby setting focusToHardLeft back to False.
 		for region in oldFocusRegions[0:commonRegionsEnd]:
+			region.focusToHardLeft = False
 			yield region
 	else:
 		# Fetch all ancestors.
 		newAncestorsStart = 1
 
+	focusToHardLeftSet = False
 	for index, parent in enumerate(ancestors[newAncestorsStart:ancestorsEnd], newAncestorsStart):
 		if not parent.isPresentableFocusAncestor:
 			continue
 		region = NVDAObjectRegion(parent, appendText=TEXT_SEPARATOR)
 		region._focusAncestorIndex = index
+		if config.conf["braille"]["focusContextPresentation"]=='hybrid' and not focusToHardLeftSet:
+			region.focusToHardLeft = True

```

Comment:
> Would this be more readable if it just did  index == newAncestorsStart ? That would avoid the need for the  focusToHardLeftSet  flag and also make it clear that we do this only for the first new ancestor.

This is not possible due to the "if not parent.isPresentableFocusAncestor:" line.